ui/app/common/patient-search/controllers/patientsListController.js

```diff
@@ -75,19 +86,59 @@ angular.module('bahmni.common.patientSearch')
             $(container).children('patient-list-spinner').hide();
         };
 
-        $scope.getHeadings = function (patients) {
-            if (patients && patients.length > 0) {
-                var headings = _.chain(patients[0])
+        $scope.getHeadings = function () {
+            if ($scope.search.activePatients && $scope.search.activePatients.length > 0) {
+                var headings = _.chain($scope.search.activePatients[0])
                     .keys()
                     .filter(function (heading) {
                         return _.indexOf(Bahmni.Common.PatientSearch.Constants.tabularViewIgnoreHeadingsList, heading) === -1;
                     })
                     .value();
+                setActiveHeadings(headings);
+            }
+        };
 
-                return headings;
+        var setActiveHeadings = function (headings) {
+            headings.map(function (heading) {
+                var newHeading = { name: heading, sortInfo: heading };
+                if (!$scope.activeHeaders.find(function (activeHeader) {
+                    return activeHeader.name == newHeading.name && activeHeader.sortInfo == newHeading.sortInfo;
+                })) {
+                    $scope.activeHeaders.push(newHeading);
+                }
+            });
+        };
+
+        $scope.sortVisiblePatientsBy = function (sortColumn) {
+            var emptyObjects = _.filter($scope.search.searchResults, function (visiblePatient) {
+                return !_.property(sortColumn)(visiblePatient);
+            });
+
+            var nonEmptyObjects = _.difference($scope.search.searchResults, emptyObjects);
+            var sortedNonEmptyObjects = _.sortBy(nonEmptyObjects, function (visiblePatient) {
+                var value = _.get(visiblePatient, sortColumn);
+                if (!isNaN(Date.parse(value))) {
+                    var parsedDate = moment(value, "DD MMMM YYYY HH:mm:ss");
```

Comment:
For consistency throughout the codebase, it's recommended to utilize the datetime format specified in the [ui/app/common/constants](https://github.com/Bahmni/openmrs-module-bahmniapps/blob/a4f22afbd988c48521cd93adf4ab36ab2b4081ca/ui/app/common/constants.js#L72).

Reply:
Done.